src/methods/properties.js

```diff
@@ -122,7 +122,14 @@ function InternalUpdatedProperty(realm: Realm, O: ObjectValue, P: PropertyKeyVal
     } else {
       generator.emitPropertyDelete(O, P);
     }
-  } else if (!desc.configurable && desc.enumerable && O === realm.$GlobalObject && desc.value !== undefined) {
+  } else if (
+    flag === "propertyCreation" &&
```

Comment:
Good catch. Thanks for checking. I'm now rewriting InternalUpdatedProperty entirely.